aten/src/ATen/native/mps/kernels/Pooling.metal

```diff
@@ -14,82 +14,82 @@ void max_pool_3d_input_iter(
     constant T* input,
     device T* output,
     device int64_t* indices,
-    constant int64_t* input_sizes,
-    constant int64_t* input_strides,
-    device int64_t* work_pooling_dim_indices,
-    constant int64_t* kernel_size,
-    constant int64_t* stride,
-    constant int64_t* padding,
-    constant int64_t* dilation) {
-  int64_t o0 = work_pooling_dim_indices[0];
-  int64_t o1 = work_pooling_dim_indices[1];
-  int64_t o2 = work_pooling_dim_indices[2];
-
-  int64_t k0 = kernel_size[0];
-  int64_t k1 = kernel_size[1];
-  int64_t k2 = kernel_size[2];
-
-  int64_t s0 = stride[0];
-  int64_t s1 = stride[1];
-  int64_t s2 = stride[2];
-
-  int64_t d0 = dilation[0];
-  int64_t d1 = dilation[1];
-  int64_t d2 = dilation[2];
-
+    constant int32_t* input_sizes,
+    constant int32_t* input_strides,
+    thread int32_t (&work_pooling_dim_indices)[3],
+    constant int32_t* kernel_size,
```

Comment:
Have you considered using vectorized types? They result in the faster code and kind of look more natural
```suggestion
    constant uint3& kernel_size,
```